Xamarin.Forms.Build.Tasks/XamlCTask.cs

```diff
@@ -44,26 +49,34 @@ public override bool Execute(out IList<Exception> thrownExceptions)
 				return true;
 			}
 
-			var resolver = new XamlCAssemblyResolver();
-			if (!string.IsNullOrEmpty(DependencyPaths))
+			var resolver = DefaultAssemblyResolver ?? new XamlCAssemblyResolver();
+			var xamlCResolver = resolver as XamlCAssemblyResolver;
+
+			if (xamlCResolver != null)
 			{
-				foreach (var dep in DependencyPaths.Split(';'))
+				if (!string.IsNullOrEmpty(DependencyPaths))
 				{
-					Logger.LogLine(3, "Adding searchpath {0}", dep);
-					resolver.AddSearchDirectory(dep);
+					foreach (var dep in DependencyPaths.Split(';'))
+					{
+						Logger.LogLine(3, "Adding searchpath {0}", dep);
+						xamlCResolver.AddSearchDirectory(dep);
+					}
 				}
-			}
 
-			if (!string.IsNullOrEmpty(ReferencePath))
-			{
-				var paths = ReferencePath.Replace("//", "/").Split(';');
-				foreach (var p in paths)
+				if (!string.IsNullOrEmpty(ReferencePath))
 				{
-					var searchpath = Path.GetDirectoryName(p);
-					Logger.LogLine(3, "Adding searchpath {0}", searchpath);
-					resolver.AddSearchDirectory(searchpath);
+					var paths = ReferencePath.Replace("//", "/").Split(';');
+					foreach (var p in paths)
+					{
+						var searchpath = Path.GetDirectoryName(p);
+						Logger.LogLine(3, "Adding searchpath {0}", searchpath);
+						xamlCResolver.AddSearchDirectory(searchpath);
+					}
 				}
 			}
+			else {
```

Comment:
I like how quickly @praeclarum understood the very special rules we have for randomly placing opening braces, and adopted it.